lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -4986,9 +4991,71 @@ LogicalResult StmtEmitter::visitStmt(InstanceOp op) {
     }
   }
 
-  ps << PP::nbsp << PPExtString(getSymOpName(op)) << " (";
+  ps << PP::nbsp << PPExtString(getSymOpName(op));
 
   ModulePortInfo modPortInfo(cast<PortList>(moduleOp).getPortList());
+  SmallVector<Value> instPortValues(modPortInfo.size());
+  op.getValues(instPortValues, modPortInfo);
+  emitInstancePortList(op, modPortInfo, instPortValues);
+
+  ps.addCallback({op, false});
+  emitLocationInfoAndNewLine(ops);
+  if (doNotPrint) {
+    ps << PP::end;
+    startStatement();
+    ps << "*/";
+    setPendingNewline();
+  }
+  return success();
+}
+
+LogicalResult StmtEmitter::visitStmt(InstanceChoiceOp op) {
+  HWModuleOp parentMod = op->getParentOfType<hw::HWModuleOp>();
+
+  std::string referenceMacroName;
+  {
+    llvm::raw_string_ostream os(referenceMacroName);
+    os << "__circt_choice_" << getVerilogModuleName(parentMod) << "_"
+       << getSymOpName(op);
```

Comment:
How to handle name conflicts with this macro?

How about this:

The instance choice op needs to be given a macro decl that it is associated with. This is purely to guarantee a unique name and for no other reason. That is then substituted in here instead of generating a potential name conflict on the fly.

To check this, there should be a test where there is a pre-existing macro decl with this name and there is no conflict in emission.